scene/resources/default_theme/default_theme_icons_builders.py

```diff
@@ -0,0 +1,77 @@
+"""Functions used to generate source files during build time
+
+All such functions are invoked in a subprocess on Windows to prevent build flakiness.
+
+"""
+
+import os
+from io import StringIO
+from platform_methods import subprocess_main
+
+
+# See also `editor/icons/editor_icons_builders.py`.
+def make_default_theme_icons_action(target, source, env):
+
+    dst = target[0]
+    svg_icons = source
+
+    icons_string = StringIO()
+
+    for f in svg_icons:
+
+        fname = str(f)
+
+        icons_string.write('\t"')
+
+        with open(fname, "rb") as svgf:
+            b = svgf.read(1)
+            while len(b) == 1:
+                icons_string.write("\\" + str(hex(ord(b)))[1:])
+                b = svgf.read(1)
+
+        icons_string.write('"')
+        if fname != svg_icons[-1]:
+            icons_string.write(",")
+        icons_string.write("\n")
+
+    s = StringIO()
+    s.write("/* THIS FILE IS GENERATED DO NOT EDIT */\n")
+    s.write("#ifndef _DEFAULT_THEME_ICONS_H\n")
+    s.write("#define _DEFAULT_THEME_ICONS_H\n")
+    s.write("static const int default_theme_icons_count = {};\n\n".format(len(svg_icons)))
+    s.write("#ifdef MODULE_SVG_ENABLED\n")
```

Comment:
You need to include `modules/modules_enabled.gen.h` to know which modules are enabled, we don't pass the values to the main env to avoid bloating the command line with too many defines.
```suggestion
    s.write('#include "modules/modules_enabled.gen.h"\n')
    s.write("#ifdef MODULE_SVG_ENABLED\n")
```
(or whatever string formatting `black` would use to allow inner quotes)